app/uk/gov/hmrc/merchandiseinbaggage/service/CalculationService.scala

```diff
@@ -54,20 +55,31 @@ class CalculationService @Inject()(mibConnector: MibConnector)(implicit ec: Exec
         )))
   }
 
-  def thresholdAllowance(maybeGoodsDestination: Option[GoodsDestination], goodsEntries: GoodsEntries)(
-    implicit hc: HeaderCarrier): OptionT[Future, ThresholdAllowance] =
+  def thresholdAllowance(
+    maybeGoodsDestination: Option[GoodsDestination],
+    goodsEntries: GoodsEntries,
+    journeyType: JourneyType,
+    declarationId: DeclarationId)(implicit hc: HeaderCarrier): OptionT[Future, ThresholdAllowance] =
     for {
       declarationGoods <- OptionT.fromOption(goodsEntries.declarationGoodsIfComplete)
       destination      <- OptionT.fromOption(maybeGoodsDestination)
-      calculation      <- OptionT.liftF(paymentCalculations(declarationGoods.goods, destination))
-    } yield ThresholdAllowance(declarationGoods, calculation, destination)
+      totalGoods       <- addGoods(journeyType, declarationId, declarationGoods.goods)
+      calculation      <- OptionT.liftF(paymentCalculations(totalGoods, destination))
+    } yield ThresholdAllowance(DeclarationGoods(totalGoods), calculation, destination)
 
   def thresholdAllowance(declaration: Declaration)(implicit hc: HeaderCarrier): Future[ThresholdAllowance] = {
     import declaration._
     paymentCalculations(declarationGoods.goods, goodsDestination).map(calculation =>
       ThresholdAllowance(declarationGoods, calculation, goodsDestination))
   }
 
+  private def addGoods(journeyType: JourneyType, declarationId: DeclarationId, goods: Seq[Goods])(
+    implicit hc: HeaderCarrier): OptionT[Future, Seq[Goods]] =
+    journeyType match {
+      case New   => OptionT.pure(goods)
+      case Amend => OptionT(findDeclaration(declarationId)).map(dec => goods ++ dec.declarationGoods.goods)
```

Comment:
I'm guessing it here, but somewhere. Amendments have to filter on `Paid` and `NotRequired` for Imports